### PR TITLE
[TEC-4696] GitHub action configs

### PR DESCRIPTION
**Pull request by sebastiandl**
### What problem is the code solving?
Configuration for Github actions.

### How does this change address the problem?
Adding new configuration file for github actions to run rspect and code quality checks. Also adding new PR template file.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        id: ruby
+        with:
+          ruby-version: ${{ env.RUBY_VERSION }}
+      - name: Install postgres client
+        run: sudo apt-get install libpq-dev
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install libcurl3-dev
+          gem install bundler -v 1.10.6
```
Comment (texpert) [inline .github/workflows/main.yml:37]: Maybe we could try a modern bundler version?

```suggestion
          gem install bundler
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        env:
+          CC_TEST_REPORTER_ID: ${{ secrets.CC_TEST_REPORTER_ID }}
+        with:
+          coverageCommand: bundler exec rake REPORT_COVERAGE=true
+          debug: true
+  code-check:
+    name:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
```
Comment (texpert) [inline .github/workflows/main.yml:53]: I think we might try to use v2 of checkout - we're already using it above, in line 22

```suggestion
    - uses: actions/checkout@v2
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      postgres:
+        image: postgres:11.6
+        ports: ["5432:5432"]
+        env:
+          POSTGRES_USER: ${{ secrets.POSTGRES_USER }}
+          POSTGRES_PASSWORD: ${{ secrets.POSTGRES_PASSWORD }}
+          POSTGRES_DB: ${{ secrets.POSTGRES_DB }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v1
```
Comment (texpert) [inline .github/workflows/main.yml:23]: We could try a modern version as well:

```suggestion
      - uses: actions/cache@v2.1.3
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+          CC_TEST_REPORTER_ID: ${{ secrets.CC_TEST_REPORTER_ID }}
+        with:
+          coverageCommand: bundler exec rake REPORT_COVERAGE=true
+          debug: true
+  code-check:
+    name:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: adwerx/pronto-ruby@v2.5
```
Comment (texpert) [inline .github/workflows/main.yml:54]:
```suggestion
    - uses: adwerx/pronto-ruby@v2.8
```

File: .github/workflows/main.yml (hunk 2 above)
Comment (sebastiandl) [reply .github/workflows/main.yml:53]: Thanks Aurel! I will make these changes you mentioned in the next push!! I honestly simply went with the same configuration we already have in mejuri-web... but if it works ok then we can also move these changes there !!

**Comment by sebastiandl** [comment]
@texpert I was not able to use the version adwerx/pronto-ruby@v2.8. There seems to be an issue when running it related to the depth of the git fetch. I tried multiple solutions I found online without success. (Eg: adding `git fetch origin master --depth=1` like mentioned in the documentation or even something like `git fetch --unshallow` that some users reported solving the problem with it).

I don't think it is worth spending too much time on that. We can eventually improve this in the near future if it is ok with you.

**Review by davidmejuri — APPROVED** [review]

**Comment by texpert** [comment]
> I don't think it is worth spending too much time on that. We can eventually improve this in the near future if it is ok with you.

Absolutely! I'd gladly get rid of both Pronto and Reek in favour of  Code Climate :)

**Review by texpert — APPROVED** [review]

**Pull request merged**